lib/services/task_lists_service.dart

```diff
@@ -0,0 +1,127 @@
+import 'dart:convert';
+
+import 'package:crdt/crdt.dart';
+import 'package:p2p_task/models/task_list.dart';
+import 'package:p2p_task/services/change_callback_provider.dart';
+import 'package:p2p_task/services/identity_service.dart';
+import 'package:p2p_task/services/sync_service.dart';
+import 'package:p2p_task/utils/key_value_repository.dart';
+import 'package:p2p_task/utils/log_mixin.dart';
+import 'package:uuid/uuid.dart';
+
+class TaskListsService with LogMixin, ChangeCallbackProvider {
+  final String _crdtTaskLisstKey = 'crdtTaskLists';
+
+  final KeyValueRepository _keyValueRepository;
+  final IdentityService _identityService;
+  final SyncService _syncService;
+
+  // stupid
+  bool isShared = false;
+
+  TaskListsService(
+    this._keyValueRepository,
+    this._identityService,
+    this._syncService,
+  );
+
+  Future<List<TaskList>> get lists async {
+    var lists = <TaskList>[];
+
+    for (var i = 0; i < (await _taskListsCrdt).values.length; i++) {
+      lists.add((await _taskListsCrdt).values[i]);
+    }
+
+    lists.sort((a, b) => a.title.toString().compareTo(b.title.toString()));
+
+    return lists;
+  }
+
+  Future upsert(TaskList taskList) async {
+    // stupid
+    if (!taskList.isShared) {
+      taskList.isShared = isShared;
```

Comment:
Will the `isShared` ever change to something other than false? And why is this checked on every upsert?

Reply:
as soon as there is one connection, it will always be true. I added "stupid" because I couldn't think about something smarter, but probably there is :D